### PR TITLE
Prevent calling RemoveStake/RemoveDelegateStake with negative amount

### PR DESCRIPTION
**Pull request by relyt29**
## What is the purpose of the change

This PR fixes a critical-severity issue that would allow any random person on the internet to denial of service the Allora chain, by sending a negative Coin amount to the RemoveDelegateStake function. The fix merely checks that the value is positive, thereby preventing an eventual panic and halt of the chain.

## Testing and Verifying

This change added tests that check that you cannot try to RemoveStake of negative amounts of coins.

## Documentation and Release Note

This PR has no implications on documentation.

### CODE REVIEW

**Comment by relyt29** [comment]
Closes ORA-1928

File: x/emissions/keeper/msgserver/msg_server_stake.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ func (ms msgServer) RemoveStake(ctx context.Context, msg *types.MsgRemoveStake)
 	if msg.Amount.IsZero() {
 		return nil, types.ErrReceivedZeroAmount
 	}
+	if msg.Amount.IsNegative() {
+		return nil, types.ErrInvalidValue
+	}
```
Comment (guilherme-brandao) [inline x/emissions/keeper/msgserver/msg_server_stake.go:67]: could be just one line

File: x/emissions/keeper/msgserver/msg_server_stake.go [hunk 2]
```diff
@@ -175,6 +178,9 @@ func (ms msgServer) RemoveDelegateStake(ctx context.Context, msg *types.MsgRemov
...
```
Comment (guilherme-brandao) [inline x/emissions/keeper/msgserver/msg_server_stake.go:183]: same

**Review by guilherme-brandao — CHANGES_REQUESTED** [review]

**Review by guilherme-brandao — APPROVED** [review]
approving but check comment plz

**Pull request merged**